reactrole/reactrole.py

```diff
@@ -85,77 +89,55 @@ async def add_reactrole(
         self,
         ctx: commands.Context,
         message: discord.Message,
-        reaction: str,
+        reaction: discord.Emoji,
         role: discord.Role,
     ):
         """Creates a new react role
 
         Example:
         - `[p]reactrole add <message id> <reaction> <role>`
         """
+        data = {"message": message.id, "reaction": str(reaction), "role": role.id, "channel": message.channel.id}
         async with self.config.guild(ctx.guild).roles() as roles:
-            added = False
+            # This should only return 1 item at max because items are checked for uniqueness before adding them
+            exists = [item for item in roles if item == data]
+            if exists:
+                return await ctx.send("React role already exists.")
 
-            for item in roles:
-                # Check if a result already exists
-                if (
-                    item["message"] == message.id
-                    and item["reaction"] == str(reaction)
-                    and item["role"] == role.id
-                    and item["channel"] == message.channel.id
-                ):
-                    added = True
-
-            if added:
-                await ctx.send("React Role already exists.")
-            else:
-                try:
-                    await message.add_reaction(str(reaction))
-
-                    roles.append(
-                        {
-                            "message": message.id,
-                            "reaction": str(reaction),
-                            "role": role.id,
-                            "channel": message.channel.id,
-                        }
-                    )
-                    await ctx.send("Configured React Role.")
-                except Exception:
-                    await ctx.send("Unable to add emoji message to message")
+            try:
+                await message.add_reaction(str(reaction))
+            except Exception:
+                return await ctx.send("Unable to add emoji message to message")
+
+            roles.append(data)
+            await ctx.send("Configured react role.")
 
     @_reactrole.command("remove")
     async def remove_reactrole(
         self,
         ctx: commands.Context,
         message: discord.Message,
-        reaction: str,
+        reaction: discord.Emoji,
         role: discord.Role,
     ):
         """Removes a configured react role
 
         Example:
-        - `[p]reactrole remove <message id> <reaction> <role>`
+        - `[p]reactrole remove 360678601227763712-893601663435276318 :kek: @moderator`
```

Comment:
Thanks Issy. Looks good from what I can tell.

Only suggestion is maybe it'd be useful to specify in help messages that the message ID should be specified as `<channel id>-<message id>` (at least that's what I'm assuming this is), or append this new example to the former example, but modify the former to something such as ``[p]reactrole remove <channel id>-<message id> <reaction> <role>``.

Thoughts?

Reply:
The bot will always generate a syntax hint for the command from the function arguments, so I feel like that would be a little redundant. This 'example' section serves only as a reminder of how those arguments would actually look in real-life usage. I think adequate information is provided there, but let me know if you think differently
![image](https://user-images.githubusercontent.com/48881813/135754622-3c2782d2-50cf-4e95-8c64-889a607f3c0a.png)

Reply:
> The bot will always generate a syntax hint for the command from the function arguments

OK, sounds fine to me then 👍